server/webapp/resources/project_scenarios.py

```diff
@@ -0,0 +1,196 @@
+import mpld3
+import json
+import uuid
+
+from flask import current_app, request
+
+from flask.ext.login import login_required
+from flask_restful import Resource, marshal_with, fields
+from flask_restful_swagger import swagger
+from flask import helpers
+
+from server.webapp.inputs import SubParser
+from server.webapp.dataio import TEMPLATEDIR, upload_dir_user
+from server.webapp.utils import (
+    load_project, load_progset, load_program, load_scenario, RequestParser, report_exception, modify_program)
+from server.webapp.exceptions import ProjectDoesNotExist, ProgsetDoesNotExist, ProgramDoesNotExist, ScenarioDoesNotExist
+from server.webapp.resources.common import file_resource, file_upload_form_parser
+from server.webapp.dbconn import db
+from server.webapp.dbmodels import ScenariosDb
+
+import optima as op
+
+scenario_parser = RequestParser()
+scenario_parser.add_arguments({
+    'name': {'type': str, 'location': 'args', 'required': True},
+    'parset_id': {'type': uuid.UUID, 'location': 'args', 'required': True},
+    'scenario_type': {'type': str, 'location': 'args', 'required': True},
+    'active': {'type': bool, 'location': 'args', 'required': True}
+})
+
+def check_pars(blob):
+    """
+    Check the pars attribute for scenarios.
+    """
+    if 'pars' not in blob.keys():
+        raise ValueError("JSON body requires 'pars' parameter")
+
+    if not isinstance(blob['pars'], list):
+        raise ValueError("'pars' needs to be a list.")
+
+    pars = []
+
+    for i in blob['pars']:
+
+        pars.append({
+            'endval': int(i['endval']),
+            'endyear': int(i['endyear']),
+            'name': str(i['name']),
+            'for': list(i['for']),
+            'startval': int(i['startval']),
+            'startyear': int(i['startyear'])
+        })
+
+    return pars
+
+# /api/project/<project-id>/scenarios
+
+class Scenarios(Resource):
+    """
+    Scenarios for a given project.
+    """
+    method_decorators = [report_exception, login_required]
+
+    @swagger.operation()
+    @marshal_with(ScenariosDb.resource_fields, envelope='scenarios')
+    def get(self, project_id):
+        """
+        Get the scenarios for the given project
+        """
+        project_entry = load_project(project_id)
+        if project_entry is None:
+            raise ProjectDoesNotExist(id=project_id)
+
+        reply = db.session.query(ScenariosDb).filter_by(project_id=project_entry.id).all()
+        return reply
+
+    @swagger.operation(
+        parameters=scenario_parser.swagger_parameters(),
+        notes="""
+            Create a new scenario.
+
+            If it is a Parameter scenario, the request body should be a JSON
+            dict with the 'pars' key.
+        """
+    )
+    @marshal_with(ScenariosDb.resource_fields)
```

Comment:
if you use `responseClass=ScenarioDb.__name__` in `swagger.operation` you don't need to use `marshal_with (because there you are using the `resource_fields` on that class)